### ISSUE TITLE
prefer-stateless-function should not check a .d.ts for a Stateful Component

### ISSUE THREAD

**Issue by Symous** [yannickcr/eslint-plugin-react#2296]
Imagine that we have a stateful react component and is about to creating a  declaration file d.ts for the .js/.jsx file. 
```
declare class StateCom extends React.Component<Props> {}
export default StateCom;
``` 
and error/warning occurs that: 
`Component should be written as a pure function eslint(react/prefer-stateless-function)`

How can we handle this since the component is a truly Component with state instead of a functional component?

**Comment by ljharb** [comment]
Indeed, d.ts files shouldn’t detect any components in them.

Our component detection probably needs to ensure there’s a class body present.

**Comment by ljharb** [comment]
cc @alexzherdev

**Comment by Symous** [comment]
@ljharb Thanks for your timely response.I am new to eslint and  that confused me a lot. Looking forward to this feature :)
